bokeh/core/property/dataspec.py

```diff
@@ -380,33 +379,31 @@ class MarkerSpec(DataSpec):
     def __init__(self, default, help=None, key_type=_ExprFieldValueTransform):
         super().__init__(key_type, MarkerType, default=default, help=help)
 
-
-class UnitsSpec(NumberSpec):
+class PropertyUnitsSpec(NumberSpec):
     """ A |DataSpec| property that accepts numeric fixed values, and also
-    serializes associated units values.
+    provides an associated units property to store units information.
 
     """
-    def __init__(self, default, units_type, units_default, help=None):
-        super().__init__(default=default, help=help, key_type=_ExprFieldValueTransformUnits)
-        self._units_type = self._validate_type_param(units_type)
 
-        # TODO (bev) units_type was already constructed, so this really should not be needed
-        self._units_type.validate(units_default)
-        self._units_type._default = units_default
-        self._units_type._serialized = False
+    def __init__(self, default, units_enum, units_default, help=None):
+        super().__init__(default=default, help=help, key_type=_ExprFieldValueTransformUnits)
+        units_type = Enum(units_enum, default=units_default, serialized=False, help=f"""
+        Units to use for the associated property: {nice_join(units_enum)}
```

Comment:
Having `_unit` consistently displayed is a very helpful addition to the reference guide! 

I would like to propose making the word 'Units' a link to https://docs.bokeh.org/en/latest/docs/user_guide/styling.html#screen-units-and-data-space-units (userguide_styling_units). My impression is that understanding the difference between data and screen units is fundamental to harnessing a lot of Bokeh's power. I think it would be very helpful to have this detailed explanation from the user guide linked right here.

Reply:
Unfortunately there are more units than just that, e.g angles also have several possible units values. That's why I listed the values at the end at least. I think anything else will have to wait for the future work I mentioned (some ability to customize the help string more than is currently possible)